wordpress-mobile/WordPress-Android
# Enable cancel autoupload action for drafts and update labels

Fixes #10465

Review instructions:
1. Turn this Draft PR into a real PR -> We haven't agreed that this is the solution we want to go with, so I created just a draft PR. Feel free to turn it into a real PR if you like the solution.
2. Add a "copy review" label
3. Review the PR
4. Update target branch to `master-auto-upload` when https://github.com/wordpress-mobile/WordPress-Android/pull/10446 is merged
5. Approve and merge the PR

Note:
Initial goal of #10465 was to handle scenario when the user adds a media to a draft in offline - allow them to cancel the auto-upload action. However, I realized we probably should allow them to cancel the auto-upload even when there isn't any failed media upload (more info https://github.com/wordpress-mobile/WordPress-Android/issues/10465#issuecomment-534040153). 
I decided to show the cancel action on all drafts which are pending auto-upload (the changes were confirmed by the user by clicking on the "Save" button). I also decided to update the labels so the UX is consistent.


To test:
1. Open an existing draft
2. Turn on airplane mode
3. Add an image
4. Click on "Save"
5. Notice the item in the post list has yellow "Couldn't perform operation. We'll try again later" label and yellow "Cancel" action.
-----
1. Open an existing draft
2. Turn on airplane mode
3. Edit the content
4. Click on "Save"
5. Notice the item has yellow "Draft will be saved to the server next time your device is online" and "Cancel" action
6. Click on cancel
7. Notice snackbar with "Changes will not be saved to the server" is shown
(this might not be ideal as the changes will actually be saved using remote-auto-save -> not sure if we should worry about it or just keep it simple)


Update release notes:

- [ x ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

## Copy Review Request

Please see https://github.com/wordpress-mobile/WordPress-Android/pull/10512#issuecomment-534782110 for the summary.
